includes/sanitizers/class-amp-accessibility-sanitizer.php

```diff
@@ -66,4 +69,89 @@ static function ( $predicate ) {
 			}
 		}
 	}
+
+	/**
+	 * Add skip link markup and style.
+	 *
+	 * @return void
+	 */
+	public function add_skip_link() {
+
+		// Early exit if not a block theme.
+		if ( ! current_theme_supports( 'block-templates' ) ) {
+			return;
+		}
+
+		// Early exit if not a block template.
+		global $_wp_current_template_content;
+		if ( ! $_wp_current_template_content ) {
+			return;
+		}
+
+		$main_tag = $this->dom->getElementsByTagName( Tag::MAIN )->item( 0 );
+		if ( ! $main_tag instanceof Element ) {
+			return;
+		}
+
+		$skip_link_target = $this->dom->getElementId( $main_tag, 'wp--skip-link--target' );
```

Comment:
Good use of `getElementId`.

Why the extra hyphens? Why not just:

```suggestion
		$skip_link_target = $this->dom->getElementId( $main_tag, 'wp-skip-link-target' );
```

Reply:
I think the double hyphens are here to match the JS implementation you'll find in Gutenberg:

https://github.com/WordPress/gutenberg/blob/33af5c0206566c8d76d17e21c11ad7635991881c/lib/compat/wordpress-5.9/templates.php#L236

Reply:
That being said, I just noticed that AMP adds a suffix to the skip link ID attribute (e.g. `wp--skip-link--target-0`). So it seems that the ID attributes won't be the same on AMP and on non-AMP versions. I guess it doesn't matter in any way.